### PR TITLE
Warn GHC 9 Compatibility to LSP Client

### PR DESCRIPTION
**Pull request by konn**
As I proposed in #1990, this PR makes ghcide to send Warning to LSP clients if GHC >= 9.0.1.
It sends the message right before the initialisation; if it's inappropriate, please fix me.

It also adds README to the notes on GHC 9 compatibility.

When invoked on GHC 9 projects, ghcide warns like this:

スクリーンショット 2021-07-01 10 47 27

expanded:

スクリーンショット 2021-07-01 10 47 31

### CODE REVIEW

File: ghcide/src/Development/IDE/GHC/Util.hs [hunk 1]
```diff
@@ -280,3 +281,10 @@ ioe_dupHandlesNotCompatible :: Handle -> IO a
 ioe_dupHandlesNotCompatible h =
    ioException (IOError (Just h) IllegalOperation "hDuplicateTo"
                 "handles are incompatible" Nothing Nothing)
+
+isOverGhc9 :: Bool
```
Comment (jneira) [inline ghcide/src/Development/IDE/GHC/Util.hs:285]: We have something similar in hls-test-utils but more general, maybe it worths move the test code to check ghc version to Development.IDE.GHC.Compat and make it available for the entire codebase:

https://github.com/haskell/haskell-language-server/blob/a0def3c4236b12105970b59f791bf86df666fbcc/hls-test-utils/src/Test/Hls/Util.hs#L110-L126

**Comment by wz1000** [comment]
I think this is not worth the annoyance on every start up. I would consider it acceptable if it was limited to the first start.

**Comment by jneira** [comment]
> 
> 
> I think this is not worth the annoyance on every start up. I would consider it acceptable if it was limited to the first start.

Is it possible using only the lsp spec? If it is not maybe the simpler thing is add the warning in the vscode editor, when it downlods hls for ghc-9.0.1. It should be done for each editor though.

**Comment by anka-213** [comment]
vscode is the only one that downloads the language server automatically, so it's probably not as important for the other editors.

**Comment by jneira** [comment]
maybe we could convert the alert in a log output, like the warning about the implicit cradle?

**Comment by konn** [comment]
> maybe we could convert the alert in a log output, like the warning about the implicit cradle?

Hmm, log output cannot be seen unless a user actively opens and navigate it so it makes less sense, I think...
Well, if there is not much consensus on adding this warning, I would close this.

**Comment by jneira** [comment]
> Hmm, log output cannot be seen unless a user actively opens and navigate it so it makes less sense, I think...
Well, if there is not much consensus on adding this warning, I would close this.

Well i can imagine a situation where users not using tha automatic download of vscode get an error or detect a plugin is not enabled (and they did not know they are disabled for ghc 9) and examine and/or post the logs in an issue. The message will help them or us to detect quickly the cause.

**Comment by jneira** [comment]
@konn do you mind i take over the pr to convert the alert in a los message? fine with closing it too if you think it is better

**Comment by konn** [comment]
Sorry for leaving this PR idle (I've been a bit busy these days...). 

> @konn do you mind i take over the pr to convert the alert in a los message?

Ah, yes please! And thanks for promoting GhcVer things to project-level.

**Review by jneira — APPROVED** [review]
lgtm 😉, thanks for adding the warning

**Pull request merged**